qemu/tests/x86_cpu_L3_cache.py

```diff
@@ -0,0 +1,63 @@
+import logging
+
+from virttest import env_process
+from virttest import error_context
+from virttest import utils_misc
+from virttest import utils_qemu
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Check L3 cache present to guest
+
+    1. boot guest with latest machine_type('pc' or 'q35'),
+     checking L3 cache presents inside guest.
+    2. Boot guest with old machine type(rhel7.3.0, pc or q35),
+     L3 cache shouldn't present inside guest.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    def boot_and_check_guest(machine_type, check_L3=False):
+        """
+        Boot guest and check L3 cache inside guest
+
+        :param machine_type: Boot guest with which machine type
+        :param check_L3: if L3 cache should exist on guest
+        """
+        params['machine_type'] = machine_type
+        params['start_vm'] = 'yes'
+        vm_name = params['main_vm']
+        L3_existence = 'present' if check_L3 else 'not present'
+        logging.info('Boot guest with machine type %s and expect L3 cache %s'
+                     ' inside guest' % (machine_type, L3_existence))
+        env_process.preprocess_vm(test, params, env, vm_name)
+        vm = env.get_vm(vm_name)
+        session = vm.wait_for_login()
+        output = session.cmd_output('lscpu')
+        session.close()
+        vm.destroy()
+        L3_present = 'L3 cache' in output
+        if check_L3 ^ L3_present:
+            test.fail('L3 cache should %s inside guest for machine type %s'
+                      ' while it is not' % (L3_existence, machine_type))
+
+    qemu_bin = utils_misc.get_qemu_binary(params)
+    machine_type = params['machine_type']
+    if machine_type == 'q35':
+        m_keyword = machine_type
+    else:
+        machine_type = 'pc'
```

Comment:
This code is redundant，we can get the default `machine_type`  in Line 49.

Reply:
Others look good to me. Thanks.

Reply:
> This code is redundant，we can get the default `machine_type` in Line 49.

No it's not, it's because I see there are many other different named pc `machine_type`s, checking subtests.cfg you can find it, So here this step is to make sure we only test `pc` no matter what user defined it for their test loop.